packages/block-editor/src/components/colors/use-colors.js

```diff
@@ -0,0 +1,96 @@
+/**
+ * External dependencies
+ */
+import memoize from 'memize';
+import { startCase } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Children, cloneElement, useMemo } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import InspectorControls from '../inspector-controls';
+import PanelColorSettings from '../panel-color-settings';
+import { useBlockEditContext } from '../block-edit';
+
+const createComponent = memoize( ( attribute, color ) => ( { children } ) =>
+	// Clone children, setting the style attribute from the color configuration,
+	// if not already set explicitly through props.
+	Children.map( children, ( child ) =>
+		cloneElement( child, {
+			style: { [ attribute ]: color, ...child.props.style },
+		} )
+	)
+);
+
+const createSetColor = memoize( ( setAttributes, name ) => ( newColor ) =>
+	setAttributes( { [ name ]: newColor } )
+);
+
+const InspectorControlsColorPanel = ( { title, colorSettings } ) => (
+	<InspectorControls>
+		<PanelColorSettings
+			title={ title }
+			initialOpen={ false }
+			colorSettings={ colorSettings }
+		/>
+	</InspectorControls>
+);
+
+export default function useColors(
```

Comment:
Should we implement this hook as experimental (at least for one release)?

Reply:
Sure

Reply:
Maybe instead of making it experimental it could be just a utility used in the block editor module for now, we don't have to expose it?

Reply:
Wouldn't it get more feedback as `experimental`?

Reply:
Are we interested in third-party feedback now? I think we should first use it in our blocks (multiple ones). I don't feel strongly though.

Reply:
>Are we interested in third-party feedback now?

Why not?